### PR TITLE
Coop Spreadsheet View - Resolves #190

### LINKED ISSUE #190: Spreadsheet View for Curation Review
### General
This view would look like our google spreadsheet, with related tables flattened out into a single row. (see below for more on this). The view is described 
1. Column Labels (field labels) at the top.  Which columns appear is yet to be determined.
2. Rows would contain flattened records from related tables
3. Scroll bars would offer vertical and horizontal navigation of the records

### Top Row (Column Label) Actions
When an admin/curator clicks on a label, they would get a choice of sorting or filtering.  

Clicking on sort would allow them to select ascending or descending. 

Clicking on filter would pop up a dropdown
1. Initially it would show all values for the field with a checkbox in front of each value plus 
5. A "Select all" choice to reselect all if they were filtered 
6. A "Clear" choice to deselect all
7. The user can or uncheck each value, and the spreadsheet displays the results

Multiple columns could be filtered with AND logic.  So, for instance the user could see all housing co-ops in Albany Park

### Content Row Actions
When a user clicks on a content row, the app pops us a dialogue box as described below
1. Consumes part of the screen in front of the spreadsheet view
2. Displays the Record for Editing (we can reuse Davey's code)
3. If necessary, the record can be scrolled vertically to get to all the fields withing the dialogue box
4. At the bottom of the dialogue box, there are buttons for "Submit" and "Cancel"

### PR DESCRIPTION
**Pull request by domdelorenzo**
Updated spreadsheet component (available at http://localhost:3001/spreadsheet in docker) now spawns an update modal window that can update Coop records. 

Outstanding question:
- How do we want to display data for the "proposed changes" column? (it's currently a json object - do we want to stringify it in the spreadsheet?)

### CODE REVIEW

**Comment by laredotornado** [comment]
Hi, Answer is, we don't.  "proposed changes" is only for people who submit
coop edit/add data for an admin to approve.  THis column should definitely
be suppressed in the spreadsheet view.

Thanks! -

File: apache/my-vhosts.conf [hunk 1]
```diff
@@ -19,6 +19,8 @@
     ProxyPassReverse /save_to_sheet_from_form/ http://web:8000/save_to_sheet_from_form/
     ProxyPass /login http://web:8000/login
     ProxyPassReverse /login http://web:8000/login
+    ProxyPass /login http://web:8000/all
+    ProxyPassReverse /login http://web:8000/all
```
Comment (laredotornado) [inline apache/my-vhosts.conf:23]: For these two lines, you'll want to change "/login" to "/all", making this

```
    ProxyPass /all http://web:8000/all
    ProxyPassReverse /all http://web:8000/all
```

File: client/src/components/Spreadsheet.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import ModalUpdate from './ModalUpdate';
+const { REACT_APP_PROXY } = process.env;
+
+export default function SpreadsheetToastGrid(){
+  const [dataArray, setDataArray] = useState([])
+  const [show, setShow] = useState(false);
+  const [selectedCoop, setSelectedCoop] = useState([])
+  const handleClose = () => {
+    setShow(false);
+    setSelectedCoop([])
```
Comment (laredotornado) [inline client/src/components/Spreadsheet.jsx:17]: Minor thing, but I notice sometimes you use semicolons at the end of lines and other times not, any specific reason?

**Review by laredotornado — COMMENTED** [review]
Great PR!  Since we already talked about excluding the "proposed_changes" from teh serializer, I'll wait to approve until you have that in.

File: client/src/components/Spreadsheet.jsx (hunk 2 above)
Comment (domdelorenzo) [reply client/src/components/Spreadsheet.jsx]: Thanks for calling that out - that's sloppy on my part. I typically use a vscode extension to auto-add semicolons at the end of lines, but it doesn't seem to be working.

File: apache/my-vhosts.conf (hunk 1 above)
Comment (domdelorenzo) [reply apache/my-vhosts.conf]: Updated, although I'm not sure how to test this one.. How do you access the proxy when it's served by docker?

**Comment by domdelorenzo** [comment]
@laredotornado, I removed "proposed_changes" from the serializer and fixed some of the sloppy formatting. I've also updated the apache config, but I'm not sure how to test it. Let me know what I should be looking at and I'll take a look!

**Comment by laredotornado** [comment]
Sadly right now, after updating the Apache configs you have to rebuild the
docker containers

```
docker-compose down --rmi all
docker-compose up
```

After that's done, you should be able to interact with your spreadsheet

```
http://loaclhost:3001/spreadsheet
```

or hit the new endpoint directly

```
http://localhost:9090/all
```

I can also give it a whirl

**Pull request merged**